pandas/io/tests/test_pickle.py

```diff
@@ -44,8 +44,15 @@ def compare_element(self, result, expected, typ, version=None):
             return
 
         if typ.startswith('sp_'):
+            # SparseTimeSeries deprecated in 0.17.0
+            if (typ == "sp_series" and version and
```

Comment:
``test_pickle`` can use object-specific method if it is defined. Defining ``compare_sp_series_ts`` (like other ``compare_xxx`` method) should allow ``test_pickle`` to use the method for sparse time series. Current impl skips type check of all sparse series.

NOTE: method name is derived from pickle key:

- https://github.com/pydata/pandas/blob/master/pandas/io/tests/generate_legacy_storage_files.py#L157